src/Microsoft.Health.Fhir.CosmosDb/Features/Health/CosmosHealthCheck.cs

```diff
@@ -142,6 +124,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                             { "Error", FhirHealthErrorCode.Error412.ToString() },
                         });
                 }
+                catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.RequestTimeout)
+                {
+                    // Handling timeout exceptions
+
+                    _logger.LogWarning(
+                        ex,
+                        "Failed to connect to the data store. Request has timed out.");
+
+                    return HealthCheckResult.Degraded(
```

Comment:
Should timeout requests to CosmosDB result in Degraded or ServiceUnavailable? 408 status code can mean the database is overloaded from client requests.

Reply:
Discussed with @fhibf - degraded is the proper behavior here.